gamestonk_terminal/cryptocurrency/coingecko/pycoingecko_view.py

```diff
@@ -0,0 +1,1570 @@
+        argparse arguments
+
+    """
+    parser = argparse.ArgumentParser(
+        add_help=False,
+        prog="market_data",
+        description="""
+                        Coin Market Data
+                        """,
+    )
+
+    try:
+        ns_parser = parse_known_args_and_warn(parser, other_args)
+
+        if not ns_parser:
+            return
+
+        df = coin.market_data
+        print(
+            tabulate(
+                df,
+                headers=df.columns,
+                floatfmt=".2f",
+                showindex=False,
+                tablefmt="fancy_grid",
+            )
+        )
+        print("")
+
+    except SystemExit:
+        print("")
+
+    except Exception as e:
+        print(e)
+        print("")
+
+
+def holdings_overview(other_args: List[str]):
+    """
+    Shows overview of public companies that holds ethereum or bitcoin from www.coingecko.com
+    Parameters
+    ----------
+    other_args: List[str]
+        Arguments to pass to argparse
+
+    """
+    parser = argparse.ArgumentParser(
+        prog="holdings_overview",
+        add_help=False,
+        description="Shows overview of public companies that holds ethereum or bitcoin",
+    )
+
+    parser.add_argument(
+        "-c",
+        "--coin",
+        dest="coin",
+        type=str,
+        help="companies with ethereum or bitcoin",
+        default="bitcoin",
+        choices=["ethereum", "bitcoin"],
+    )
+
+    try:
+        ns_parser = parse_known_args_and_warn(parser, other_args)
+        if not ns_parser:
+            return
+
+        df = gecko.get_holdings_overview(endpoint=ns_parser.coin)
+        print(
+            tabulate(
+                df,
+                headers=df.columns,
+                floatfmt=".2f",
+                showindex=False,
+                tablefmt="fancy_grid",
+            )
+        )
+        print("")
+
+    except Exception as e:
+        print(e)
+        print("")
+
+
+def holdings_companies_list(other_args: List[str]):
+    """Shows Ethereum/Bitcoin Holdings by Public Companies from www.coingecko.com
+    Track publicly traded companies around the world that are buying ethereum as part of corporate treasury
+
+    Parameters
+    ----------
+    other_args: List[str]
+        Arguments to pass to argparse
+
+    """
+    parser = argparse.ArgumentParser(
+        prog="holdings_companies_list",
+        add_help=False,
+        description="Track publicly traded companies around the world that "
+        "are buying ethereum as part of corporate treasury",
+    )
+
+    parser.add_argument(
+        "-c",
+        "--coin",
+        dest="coin",
+        type=str,
+        help="companies with ethereum or bitcoin",
+        default="bitcoin",
+        choices=["ethereum", "bitcoin"],
+    )
+
+    try:
+        ns_parser = parse_known_args_and_warn(parser, other_args)
+        if not ns_parser:
+            return
+
+        df = gecko.get_companies_assets(endpoint=ns_parser.coin)
```

Comment:
Did you do something different here?

Asking because the table shows like this:
<img width="1177" alt="Captura de ecrã 2021-06-28, às 02 25 53" src="https://user-images.githubusercontent.com/25267873/123566729-4e260180-d7b8-11eb-8b98-6180dbd35c63.png">

Where in 1 hand the table doesn't overrun the screen, but in the other hand the link is not cliable anymore as there are new lines in betwen.

Reply:
Yes, in some data frames which has big width I added a method, that wraps long text into new line, base on given len. So this way I could fit width of displayed table into terminal width. But question, what to you think it's better idea for that ? 
1. Remove all the urls from tables (issue with too-long columns will be fixed)
2. Like in you example, just split to long text into new lines (inside column)
What do you think ?

Reply:
So, my opinion on that is that:
1. We want the table to be displayed nicely to the user
2. We want the link to be clickable

In this case you solve 1 at the expense of 2. Which works, but the issue is that the link is no longer clicable, and if you want to access it you need to copy-paste for as many lines as the link contains, which is not ideal.

Hence, due to that I would say to not display columns by default, and have a flag to "-l" or "--links"  and if the user selects such bool flag, we would display a new dataframe with columns Rank, Company, Url.

What do you think?

This is just myself putting me on the user side of things, and thinking about what I would find useful

Reply:
Yea, agree. From user perspective it should be clicable and pretty. So the idea for views that display urls, which are very long should have this kind of flag (probably there will be few more views which can have simillar behaviour). But let's try with --links flag
Thank you for advice